easybuild/easyblocks/p/python.py

```diff
@@ -192,6 +192,19 @@ def patch_step(self, *args, **kwargs):
             # Ignore user site dir. -E ignores PYTHONNOUSERSITE, so we have to add -s
             apply_regex_substitutions('configure', [(r"(PYTHON_FOR_BUILD=.*-E)'", r"\1 -s'")])
 
+        # If we filter out LD_LIBRARY_PATH (not unusual when using rpath), ctypes is not able to dynamically load
+        # libraries installed with EasyBuild (see https://github.com/EESSI/software-layer/issues/192).
+        # Allow ctypes to also use LIBRARY_PATH in this scenario (preferring the standard LD_LIBRARY_PATH)
+        filtered_env_vars = build_option('filter_env_vars') or []
+        if 'LD_LIBRARY_PATH' in filtered_env_vars and 'LIBRARY_PATH' not in filtered_env_vars:
+            ctypes_util_py = os.path.join("Lib", "ctypes", "util.py")
+            orig_ld_libs = "libpath = os.environ.get('LD_LIBRARY_PATH')"
+            orig_ld_libs_regex = r'(\s*)' + re.escape(orig_ld_libs) + r'(\s*)'
+            updated_ld_libs = ("libpath = ':'.join(filter(None, "
+                               "os.environ.get('LD_LIBRARY_PATH', '').split(':') + "
+                               "os.environ.get('LIBRARY_PATH', '').split(':')))")
+            apply_regex_substitutions(ctypes_util_py, [(orig_ld_libs_regex, r'\1' + updated_ld_libs + r'\2')])
+
```

Comment:
@trz42 @boegel I'd like to add a sanity check command that would catch this, but I've tried to do something in EESSI and can't seem to figure out the case where it is broken. I tried to use the library `libffi.so` since that appears as a dependency (and thus is in `LIBRARY_PATH`) but this succeeded when I thought it should fail. Any ideas?
```
python -c "import ctypes;  testlib = ctypes.cdll.LoadLibrary('libffi.so')"
```

Reply:
The python commands that failed were a bit different, maybe you can try those (or use the pattern but `ffi` as test library)?

```bash
python -c "from ctypes.util import find_library; from ctypes import CDLL; find_library('mpi'); CDLL(find_library('mpi'))"
```

Reply:
Urgh, I am wrong here, all this does is help the `ld` to find the library, but we have no problem finding the library as it tries to use `gcc` to find it before falling back to this. This doesn't solve the _actual_ issue that is in  #192

Reply:
Changed approach with https://github.com/easybuilders/easybuild-easyblocks/pull/3352/commits/d07f920031f4431b388318895e308a9e7b20d9b2